client/client.go

```diff
@@ -136,33 +177,24 @@ func New(config *Config) (*Client, error) {
 		config = &Config{}
 	}
 
-	var client *Client
-	var transport *http.Transport
-
-	if config.BaseURL == "" {
-		// By default talk over a unix socket.
-		transport = &http.Transport{Dial: unixDialer(config.Socket), DisableKeepAlives: config.DisableKeepAlive}
-		baseURL := url.URL{Scheme: "http", Host: "localhost"}
-		client = &Client{baseURL: baseURL}
-	} else {
-		// Otherwise talk regular HTTP-over-TCP.
-		baseURL, err := url.Parse(config.BaseURL)
-		if err != nil {
-			return nil, fmt.Errorf("cannot parse base URL: %v", err)
-		}
-		transport = &http.Transport{DisableKeepAlives: config.DisableKeepAlive}
-		client = &Client{baseURL: *baseURL}
+	client := &Client{}
+	requester, err := newDefaultRequester(client, config)
+	if err != nil {
+		return nil, err
 	}
 
-	client.doer = &http.Client{Transport: transport}
-	client.userAgent = config.UserAgent
+	client.requester = requester
 	client.getWebsocket = func(url string) (clientWebsocket, error) {
-		return getWebsocket(transport, url)
+		return getWebsocket(requester.Transport(), url)
 	}
 
 	return client, nil
 }
 
+func (client *Client) Requester() Requester {
+	return client.requester
+}
+
 func (client *Client) getTaskWebsocket(taskID, websocketID string) (clientWebsocket, error) {
 	url := fmt.Sprintf("ws://localhost/v1/tasks/%s/websocket/%s", taskID, websocketID)
```

Comment:
I think it may make sense to do this as separate exercise. The client code around dealing with baseURL/socket path and the server code dealing with socket path / HTTP address seems to be involved. What I propose is that we look at this when we work on a PR to enable TCP client comms. Will that be OK ?

Reply:
Confirmed with Angel. Resolving.